.github/workflows/tests.yaml

```diff
@@ -0,0 +1,26 @@
+name: Run Unit Tests
+
+on: pull_request
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version:
+          - "3.11"
+          - "3.12"
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: pip install .
+      - name: Run tests
+        run: |
+          pytest test/ # or use unittest, e.g., python -m unittest discover
```

Comment:
There are markers defined in `pyproject.toml` but this is reasonable for now since we don't need to be particular about what tests get executed here yet